L1Trigger/L1TTrackMatch/plugins/L1TrackVertexAssociationProducer.cc

```diff
@@ -0,0 +1,449 @@
+         std::vector<TTTrack> - A collection of TTTracks selected from cuts on the TTTrack_TrackWord properties
+*/
+//
+// Original Author:  Alexx Perloff
+//         Created:  Thu, 16 Dec 2021 19:02:50 GMT
+// Derivative Author: Nick Manganelli
+//         Created: Thu, 16 Feb 2023 16:03:32 GMT
+//
+//
+
+// system include files
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+// Xilinx HLS includes
+#include <ap_fixed.h>
+#include <ap_int.h>
+
+// user include files
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/Common/interface/RefVector.h"
+#include "DataFormats/Common/interface/RefToPtr.h"
+#include "DataFormats/Common/interface/Ptr.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+#include "DataFormats/L1Trigger/interface/Vertex.h"
+#include "DataFormats/L1Trigger/interface/VertexWord.h"
+#include "L1Trigger/VertexFinder/interface/L1Track.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "CommonTools/Utils/interface/AndSelector.h"
+#include "CommonTools/Utils/interface/EtaRangeSelector.h"
+#include "CommonTools/Utils/interface/MinSelector.h"
+#include "CommonTools/Utils/interface/MinFunctionSelector.h"
+#include "CommonTools/Utils/interface/MinNumberSelector.h"
+#include "CommonTools/Utils/interface/PtMinSelector.h"
+#include "CommonTools/Utils/interface/Selection.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+
+//
+// class declaration
+//
+
+class L1TrackVertexAssociationProducer : public edm::global::EDProducer<> {
+public:
+  explicit L1TrackVertexAssociationProducer(const edm::ParameterSet&);
+  ~L1TrackVertexAssociationProducer() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  // ----------constants, enums and typedefs ---------
+  // Relevant constants for the converted track word
+  enum TrackBitWidths {
+    kPtSize = TTTrack_TrackWord::TrackBitWidths::kRinvSize - 1,  // Width of pt
+    kPtMagSize = 9,                                              // Width of pt magnitude (unsigned)
+    kEtaSize = TTTrack_TrackWord::TrackBitWidths::kTanlSize,     // Width of eta
+    kEtaMagSize = 3,                                             // Width of eta magnitude (signed)
+  };
```

Comment:
These are directly from L1TrackSelectionProducer.cc (from which the TVA Producer has been split off to match the firmware architecture). These also mirror the storage of these constants in the firmware, and across all these GTT modules in CMSSW, which makes me hesitant to change the format in only one place. At some point, consolidating these enum constants and potentially changing how they're stored should indeed be considered (because it's very WET when we should be DRY). However, some discussion and approval in GTT/L1T should happen first, I think